### PR TITLE
get_upstream_artifacts_full_paths_per_task_id(): Add test to ensure s…

### PR DESCRIPTION
**Pull request by JohanLorenzo**
…ome edge cases are covered

While working on https://github.com/mozilla-releng/beetmoverscript/pull/163, I had a doubt on whether `get_upstream_artifacts_full_paths_per_task_id()` was able to handle payloads like:

```json
{
  "upstreamArtifacts": [{
    "taskId": "one",
    "paths": ["some_path"]
  }, {
    "taskId": "one",
    "paths": ["some_other_path"]
  }]
}
```

It turns out it can! I updated the test to cover this kind of use case + 2 others.

### CODE REVIEW

File: scriptworker/test/test_artifacts.py [hunk 1]
```diff
@@ -215,13 +215,24 @@ def buildSignedUrl(*args, **kwargs):
...
-        'paths': ['public/file_b'],
+        'paths': ['public/file_b1', 'public/file_b2'],
         'taskId': 'dependency2',
         'taskType': 'signing',
+    }, {
+        'paths': ['some_other_folder/file_c'],
+        'taskId': 'dependency3',
+        'taskType': 'signing',
+    }, {
+        # Case where the same taskId was given. In some occasion we may want to split
```
Comment (MihaiTabara) [inline scriptworker/test/test_artifacts.py]: Hah, this is neat, wasn't sure this is possible.

File: scriptworker/test/test_artifacts.py [hunk 2]
```diff
@@ -241,14 +252,28 @@ def test_get_upstream_artifacts_full_paths_per_task_id(context):
...
     for artifact in artifacts_to_succeed:
         folder = os.path.join(context.config['work_dir'], 'cot', artifact['taskId'])
-        os.makedirs(os.path.join(folder, 'public'))
-        touch(os.path.join(folder, artifact['paths'][0]))
+
+        for path in artifact['paths']:
+            try:
+                os.makedirs(os.path.join(folder, os.path.dirname(path)))
+            except FileExistsError:
+                pass
```
Comment (MihaiTabara) [inline scriptworker/test/test_artifacts.py]: 👍
